user_token/service.py

```diff
@@ -13,44 +13,46 @@ class UserData:
     def __init__(self, user_id, last_used_timestamp: datetime):
         self.user_id = user_id
         self.last_used_timestamp = last_used_timestamp
-    
-    
-def generate_token() -> str:
+
+
+def generate_token() -> bytes:
     """Do not use, internal function for user token generation"""
-    return str(secrets.token_bytes(USER_TOKEN_LENGTH))
+    return secrets.token_bytes(USER_TOKEN_LENGTH)
 
-def register_new_token(user_id: str) -> str:
+def register_new_token(user_id: str) -> bytes:
     """Register a token to a user who has been authenticated"""
-    if user_to_token_map.get(user_id) is not None:
-        raise Exception("This user is already registered to a token")
-    
+    existing_token = user_to_token_map.get(user_id)
+    if existing_token is not None:
+        refresh_token(existing_token)
+        return existing_token
```

Comment:
This was something I previously brought up in our discussions but that we never resolved. As a refresher on the motivations:
1. If a user logs on with another device they shouldn't have to log out on the previous device
2. There are all sorts of ways that the front-end can get reset (and the token consequently lost) without giving the user/app a chance to log out so I don't think it is reasonable to expect a user to log out after every session.